src/Ide/Cradle.hs

```diff
@@ -904,3 +912,14 @@ cradleDisplay cradle = fromString result
   name = Bios.actionName (Bios.cradleOptsProg cradle)
 
 -- ---------------------------------------------------------------------
+-- | Wrapper around 'readCreateProcess' that sets the working directory
+readProcessWithCwd :: FilePath -> FilePath -> [String] -> String -> IO (CradleLoadResult String)
+readProcessWithCwd dir cmd args stdi = do
+  let createProc = (proc cmd args) { cwd = Just dir }
+  mResult <- optional $ readCreateProcessWithExitCode createProc stdi
+  case mResult of
+    Just (ExitSuccess, stdo, _) -> pure $ CradleSuccess stdo
+    Just (exitCode, stdo, stde) -> pure $ CradleFail $
+      CradleError [] exitCode ["Error when calling " <> cmd <> " " <> unwords args, stdo, stde]
+    Nothing -> pure $ CradleFail $
+      CradleError [] ExitSuccess ["Couldn't execute " <> cmd <> " " <> unwords args]
```

Comment:
This should be exposed in hie-bios, see https://github.com/mpickering/hie-bios/pull/227